### PR TITLE
Move o.e.jface.tests to Java 17

### PR DESCRIPTION
**Pull request by vrubezhny**

### CODE REVIEW

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java [hunk 1]
```diff
@@ -9,12 +9,10 @@
  * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
- *     R�diger Herrmann - initial API and implementation
+ *     R�diger Herrmann - initial API and implementation
```
Comment (akurtakov) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java]: Name broke.

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java [hunk 2]
```diff
@@ -62,12 +62,12 @@ public void testEqualsAndHashCode() {
 				new ImageDescriptor[] { overlayDescriptor1, overlayDescriptor2 });
 		DecorationOverlayIcon icon2 = new DecorationOverlayIcon(baseImage1,
 				new ImageDescriptor[] { overlayDescriptor1, overlayDescriptor2 });
-		assertTrue(icon1.equals(icon2));
+		assertEquals(icon1, icon2);
```
Comment (akurtakov) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java]: This test is testing equals implementation so these should stay calling equals.

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java [hunk 3]
```diff
@@ -142,7 +142,7 @@ public void testEqualsAndHashCode2() {
 		// first verify image descriptor's equals/hashcode
 		SimpleImageDescriptor equalButDifferent1 = new SimpleImageDescriptor("pretend_file_name");
 		SimpleImageDescriptor equalButDifferent2 = new SimpleImageDescriptor("pretend_file_name");
-		assertTrue(equalButDifferent1.equals(equalButDifferent2));
+		assertEquals(equalButDifferent1, equalButDifferent2);
```
Comment (akurtakov) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java]: same

**Review by akurtakov — COMMENTED** [review]
Please make sure that equals tests still stay calling equals.

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java (hunk 1 above)
Comment (vrubezhny) [reply tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java]: OK

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java (hunk 2 above)
Comment (vrubezhny) [reply tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java]: OK

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java (hunk 3 above)
Comment (vrubezhny) [reply tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/DecorationOverlayIconTest.java]: OK

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java [hunk 4]
```diff
@@ -9,12 +9,10 @@
  * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
- *     R�diger Herrmann - initial API and implementation
+ *     Rüdiger Herrmann - initial API and implementation
```
Comment (akurtakov) [inline tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java]: Broken name

**Review by akurtakov — CHANGES_REQUESTED** [review]

File: tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java (hunk 4 above)
Comment (vrubezhny) [reply tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java]: Is it?

Comment (akurtakov) [reply tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/dialogs/StatusDialogTest.java]: yes, preserve it please

**Comment by akurtakov** [comment]
And rebase on master

**Comment by vrubezhny** [comment]
Rebased

**Review by akurtakov — APPROVED** [review]

**Pull request merged**